grade.go

```diff
@@ -206,30 +205,43 @@ func createCsv(results util.SubmissionResults, outfile string) {
 
 // Compiles a student submission located in $dir.
 // Returns true if compiled without errors.
-func compile(dir string, wall bool) bool {
-	compilePath, err := filepath.Glob(dir + "/*.cpp")
-	util.Throw(err)
-
+func compile(dir, language string, wall bool) bool {
+	var err error
+	var compilePath []string
 	var cmd *exec.Cmd
-	if wall {
-		compilePath = append([]string{"-Wall"}, compilePath...)
-		cmd = exec.Command("g++", compilePath...)
+	if language == "java" {
+		compilePath, err = filepath.Glob(dir + "/*.java")
+		cmd = exec.Command("javac", compilePath...)
+	} else if language == "c++" {
+		compilePath, err = filepath.Glob(dir + "/*.cpp")
+		if wall {
+			compilePath = append([]string{"-Wall"}, compilePath...)
+			cmd = exec.Command("g++", compilePath...)
+		} else {
+			cmd = exec.Command("g++", compilePath...)
+		}
 	} else {
-		cmd = exec.Command("g++", compilePath...)
+		util.Throw(err)
```

Comment:
this will never throw, err is set to `nil` in the outer scope and is only altered in branches that are mutually exclusive to this one.

Reply:
Noted. Going to fix